### PR TITLE
LIME-435: Add privacy policy to Profile Settings and Public Profile

### LINKED ISSUE #435: Enhancement: Public profile page should optionally contain private info - weight, height, age
User should be provided a possibility to choose whether he/she wants to share personal info on Public progile page. E.g. check box for agreement to share information about age, weight, height may be included on Personal information page

### LINKED ISSUE #452: BUG: User can't save valid date of birth on Personal information page
### Describe your issue

User can't save valid date of birth on Profile information page: server error message is displayed, status code 422 is received on backend

### Steps to reproduce

1. Sign up on https://live-in-motion-everyday.com/sign-up
2. Navigate to Personal Information page, clicking on gear ion in header
3. Enter valid past date of birth
4. Pay attention to the error message displayed and server response

### What was the actual result?

User can't save valid date of birth on Profile information page: server error message is displayed to user. In response status code 422 is received  and error message {
    "details": [
        {
            "message": "Invalid date format. Use DD/MM/YYYY",
            "path": [
                "dateOfBirth"
            ]
        },
        {
            "message": "Date of birth can not be in the future",
            "path": [
                "dateOfBirth"
            ]
        }
    ],
    "errorType": "VALIDATION",
    "message": "[\n  {\n    \"validation\": \"regex\",\n    \"code\": \"invalid_string\",\n    \"message\": \"Invalid date format. Use DD/MM/YYYY\",\n    \"path\": [\n      \"dateOfBirth\"\n    ]\n  },\n  {\n    \"code\": \"custom\",\n    \"message\": \"Date of birth can not be in the future\",\n    \"path\": [\n      \"dateOfBirth\"\n    ]\n  }\n]"
}

### What was the expected result?

Data is saved, Profile information page is displayed with saved value of date of birth

### Put here any screenshots or videos (optional)

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/123622110/8087a61c-34a7-42e4-9041-433d4519d144

### What OS are you seeing the problem on?

Windows

### What browsers are you seeing the problem on?

Chrome

### Severity

Major (S3)

### PR DESCRIPTION
**Pull request by valery-chumak**
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/72942456/17ecb81b-fcfd-4158-91ac-f8666418d488

### CODE REVIEW

File: frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx [hunk 1]
```diff
@@ -147,16 +149,24 @@ const ProfileSettings: React.FC<Properties> = ({
...
         setIsOpen(false);
     }, []);
 
+    const handleCheckboxChange = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>): void => {
+            setValue('isPublic', event.target.checked);
+        },
+        [setValue],
+    );
+
```
Comment (stpalamar) [inline frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx]: Do we need this callback for handling the checkbox? I guess, it is possible to handle it inside the checkbox component and set the value for a field

Comment (valery-chumak) [reply frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx]: I tried to find a way to control the checkbox value without the handler, but it was unsuccessful for me.
I tried the variant with {..register} but it didn't work.
Maybe if you have some suggestions, I would be really thankful.

Comment (stpalamar) [reply frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx]: I pushed suggestion)

**Review by stpalamar — DISMISSED** [review]

**Review by stpalamar — APPROVED** [review]

**Review by dimapopovych — APPROVED** [review]
not entirely correct approach since users data is still public and being sent to client, but acceptable in current conditions :)

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**